azurerm/internal/services/synapse/synapse_workspace_resource.go

```diff
@@ -642,3 +660,14 @@ func flattenIdentityControlSQLSettings(settings synapse.ManagedIdentitySQLContro
 
 	return false
 }
+
+func flattenEncryptionDetails(encryption *synapse.EncryptionDetails) string {
```

Comment:
Maybe have this return a pointer instead of a string so you can check for `nil`?
```suggestion
func flattenEncryptionDetails(encryption *synapse.EncryptionDetails) *string {
```